### ISSUE TITLE
RollingUpgrade resource errors with no obvious reason why

### ISSUE THREAD

**Issue by bodgit** [keikoproj/upgrade-manager#194]
**Is this a BUG REPORT or FEATURE REQUEST?**:

Bug report.

**What happened**:

After removing the attributes in #193 I got the controller running and added my first `RollingUpgrade` resource:
```
apiVersion: upgrademgr.keikoproj.io/v1alpha1
kind: RollingUpgrade
metadata:
  generateName: rollingupgrade-sample-
spec:
  asgName: mdainty-IxKAYGsd-020210219144643638800000011
  nodeIntervalSeconds: 300
  postDrain:
    waitSeconds: 90
```

Looking at the logs, it immediately marks the resource as errored:
```
2021-03-17T12:09:07.924Z        INFO    controllers.RollingUpgrade      Default strategy settings applied.      {"rollingupgrade": "rollingupgrade-sample-rvjs6", "updateStrategy": {"type":"randomUpdate","mode":"lazy","maxUnavailable":1,"drainTimeout":-1}}
2021-03-17T12:09:07.924Z        INFO    controllers.RollingUpgrade      Adding obj to map:      {"rollingupgrade": "rollingupgrade-sample-rvjs6", "name": "rollingupgrade-sample-rvjs6/kube-system"}
2021-03-17T12:09:07.924Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "rollingupgrade", "request": "kube-system/rollingupgrade-sample-rvjs6"}
2021-03-17T12:09:08.070Z        INFO    controllers.RollingUpgrade      Marked object as        {"rollingupgrade": "rollingupgrade-sample-rvjs6", "finalStatus": "error"}
2021-03-17T12:09:08.070Z        INFO    controllers.RollingUpgrade      Failed to calculate totalProcessingTime {"rollingupgrade": "rollingupgrade-sample-rvjs6"}
2021-03-17T12:09:08.086Z        INFO    controllers.RollingUpgrade      Default strategy settings applied.      {"rollingupgrade": "rollingupgrade-sample-rvjs6", "updateStrategy": {"type":"randomUpdate","mode":"lazy","maxUnavailable":1,"drainTimeout":-1}}
2021-03-17T12:09:08.086Z        INFO    controllers.RollingUpgrade      Found obj in map:       {"rollingupgrade": "rollingupgrade-sample-rvjs6", "name": "rollingupgrade-sample-rvjs6/kube-system"}
2021-03-17T12:09:08.086Z        INFO    controllers.RollingUpgrade      Object already being processed  {"rollingupgrade": "rollingupgrade-sample-rvjs6", "name": "rollingupgrade-sample-rvjs6/kube-system"}
2021-03-17T12:09:08.087Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "rollingupgrade", "request": "kube-system/rollingupgrade-sample-rvjs6"}
2021-03-17T12:09:08.087Z        INFO    controllers.RollingUpgrade      Deleted the entries of ASG in the cluster store {"rollingupgrade": "rollingupgrade-sample-rvjs6", "asgName": "mdainty-IxKAYGsd-020210219144643638800000011"}
2021-03-17T12:09:08.087Z        INFO    controllers.RollingUpgrade      Deleted from admission map      {"rollingupgrade": "rollingupgrade-sample-rvjs6", "admissionMap": {}}
```
There's nothing in those logs that tells me *why* it's errored. It's clearly already running at a debug logging level so I don't think I can make it log more. I'm now left scratching my head why it's not working.

The ASG definitely exists as if I use an obviously wrong ASG name I get an error "No ASG found with name ..." so it suggests the API calls to describe the ASG are working.

**What you expected to happen**:

Some sort of explanation as to why there's an error.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- rolling-upgrade-controller version: latest release as of this issue
- Kubernetes version : EKS 1.17

**Other debugging information (if applicable)**:
- RollingUpgrade status:
```
$ kubectl describe ru rollingupgrade-sample-nm98q
Name:         rollingupgrade-sample-nm98q
Namespace:    kube-system
Labels:       <none>
Annotations:  <none>
API Version:  upgrademgr.keikoproj.io/v1alpha1
Kind:         RollingUpgrade
Metadata:
  Creation Timestamp:  2021-03-17T14:03:01Z
  Generate Name:       rollingupgrade-sample-
  Generation:          1
  Resource Version:    6039982
  Self Link:           /apis/upgrademgr.keikoproj.io/v1alpha1/namespaces/kube-system/rollingupgrades/rollingupgrade-sample-nm98q
  UID:                 98c081ef-3363-4be1-8ce5-3615389f91a9
Spec:
  Asg Name:               mdainty-IxKAYGsd-020210219144643638800000011
  Node Interval Seconds:  300
  Post Drain:
    Wait Seconds:  90
Status:
  Conditions:
    Status:        True
    Type:          Complete
  Current Status:  error
  End Time:        2021-03-17T14:03:01Z
Events:
  Type     Reason                  Age   From             Message
  ----     ------                  ----  ----             -------
  Warning  RollingUpgradeFinished  42s   upgrade-manager  {"asgName":"mdainty-IxKAYGsd-020210219144643638800000011","info":"Rolling Upgrade as finished (status=error)","status":"error","strategy":"randomUpdate"}
  Normal   RollingUpgradeStarted   42s   upgrade-manager  {"asgName":"mdainty-IxKAYGsd-020210219144643638800000011","msg":"Rolling Upgrade has started","status":"started","strategy":"randomUpdate"}
```

**Comment by bodgit** [comment]
So this turned out to be a lack of the `ec2:DescribeLaunchTemplates` permission. I will send a PR to update the README as the policy there is incomplete, and that is what I copied. There's also a bug that is not creating an error with sufficient information to point to this lack of permission.

**Comment by shreyas-badiger** [comment]
@bodgit How were you able to figure `ec2:DescribeLaunchTemplates` permission was missing? 
Can you consider sending out a PR for the info log? Also, consider updating it in controller-v2 as well.

**Comment by bodgit** [comment]
> How were you able to figure ec2:DescribeLaunchTemplates permission was missing?

A bit of guesswork, and tracing through the code starting with this function as it seemed to be doing most of the work:

https://github.com/keikoproj/upgrade-manager/blob/4da0472228a3e4b1579d7e6285eefb8600f688f1/controllers/rollingupgrade_controller.go#L660

I then got to the `populateLaunchTemplates()` function call and realised I didn't have permission for `ec2:DescribeLaunchTemplates` in the IAM policy. Added it and things started working.

> Can you consider sending out a PR for the info log? Also, consider updating it in controller-v2 as well.

I would if I could figure out why the error isn't making it's way into the log in the first place:

https://github.com/keikoproj/upgrade-manager/blob/4da0472228a3e4b1579d7e6285eefb8600f688f1/controllers/rollingupgrade_controller.go#L489-L501

This is the only place I can find that uses the `ec2:DescribeLaunchTemplates` call. I would assume `err` should have been some form of "you don't have permission to call ec2:DescribeLaunchTemplates" error and been returned but it's been swallowed up somewhere, although I've not really tried to reproduce it.
